zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/job/DeactivateTimeOutJobs.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.engine.processing.job;
+
+import static io.camunda.zeebe.scheduler.clock.ActorClock.currentTimeMillis;
+
+import io.camunda.zeebe.engine.state.immutable.JobState;
+import io.camunda.zeebe.engine.state.immutable.JobState.DeadlineIndex;
+import io.camunda.zeebe.protocol.record.intent.JobIntent;
+import io.camunda.zeebe.stream.api.ReadonlyStreamProcessorContext;
+import io.camunda.zeebe.stream.api.scheduling.Task;
+import io.camunda.zeebe.stream.api.scheduling.TaskResult;
+import io.camunda.zeebe.stream.api.scheduling.TaskResultBuilder;
+import java.time.Duration;
+import org.agrona.collections.MutableInteger;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+final class DeactivateTimeOutJobs implements Task {
```

Comment:
🔧 I suggest we rename this to `JobTimeoutChecker` as it fully takes the role of a checker, including rescheduling itself. Let's then also rename the current `JobTimeoutChecker` to something else, as it only manages the lifecycle of the checker now.

Reply:
I also had the same feeling but I couldn't know what to rename the actual class to, so thought of keeping the current names as is.
Now I'm thinking the other one can be renamed to `JobTimeoutListner` or `JobTimeoutObserver `, WDYT?